Nodes/HTTPSyncNode.cpp

```diff
@@ -0,0 +1,284 @@
+#include "HTTPSyncNode.h"
+#include "Messages/LocalConfigChangeMsg.h"
+#include "Messages/LocalWaypointChangeMsg.h"
+#include "Messages/ServerConfigsReceivedMsg.h"
+#include "Messages/ServerWaypointsReceivedMsg.h"
+
+/*
+
+    PURPOSE:
+        HANDLE RETRIEVAL AND WRITING TO DATABASE AUTONOMOUSLY UPON LOCAL OR SERVERSIDE CHANGES
+        PUSH LOGS ON AN INTERVAL
+        NOTIFY MESSAGEBUS WHEN NEW SERVERDATA ARRIVES
+
+*/
+
+size_t write_to_string(void *ptr, size_t size, size_t count, void *stream) {
+    ((std::string*)stream)->append((char*)ptr, 0, size*count);
+    return size*count;
+}
+
+HTTPSyncNode::HTTPSyncNode(MessageBus& msgBus, DBHandler *db, int delay, bool removeLogs)
+	:ActiveNode(NodeID::HTTPSync, msgBus), m_removeLogs(removeLogs), m_delay(delay), m_dbHandler(db)
+{
+    
+}
+
+//TODO - DESTRUCTOR?
+
+bool HTTPSyncNode::init()
+{
+
+    m_initialised = false;
+
+    //previously in constructor
+    curl = curl_easy_init();
+    m_reportedConnectError = false;
+
+    m_pushOnlyLatestLogs = m_dbHandler->retrieveCellAsInt("httpsync_config", "1", "push_only_latest_logs");
+
+    try {
+        setShipID( m_dbHandler->retrieveCell("server", "1", "boat_id") );
+        setShipPWD( m_dbHandler->retrieveCell("server", "1", "boat_pwd") );
+        setServerURL( m_dbHandler->retrieveCell("server", "1", "srv_addr") );
+
+        m_initialised = true;
+
+    } catch (const char * error) {
+        Logger::error("%s Failed to get server configuration Error: %s", __PRETTY_FUNCTION__, error);
+        //Kill thread if setup fails
+        std::terminate();
+    }
+    Logger::info("HTTPSyncNode init() successful");
```

Comment:
This will be done in NodeInitialise in main, that way we keep a nice consist logging